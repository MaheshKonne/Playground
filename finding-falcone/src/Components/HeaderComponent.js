import React, { Fragment} from 'react';
import theme from '../theme';

const HeaderComponent = () => (
    <Fragment>
        <header>
            <div className='title'>Finding Falcone!</div>
        </header>
        <style>
            {`
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: ${theme.headerBackground};
            }
            `}
        </style>
    </Fragment>
);

export default HeaderComponent;