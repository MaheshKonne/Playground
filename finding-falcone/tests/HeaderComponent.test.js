import React from 'react';
import HeaderComponent from '../src/Components/HeaderComponent';
import { shallow } from "enzyme";

describe('<HeaderComponent />', function () {
    const component = shallow(<HeaderComponent/>);

    it('should match with snapshot', function () {
        global.fetch = jest.fn();
        expect(component).toMatchSnapshot();
    });
});