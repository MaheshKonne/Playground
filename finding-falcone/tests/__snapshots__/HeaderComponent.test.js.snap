// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> should match with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <header>
          <div
            className="title"
          >
            Finding Falcone!
          </div>
        </header>,
        <style>
          
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: rgba(200,200,200,0.5);
            }
            
        </style>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="title"
          >
            Finding Falcone!
          </div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Finding Falcone!",
            "className": "title",
          },
          "ref": null,
          "rendered": "Finding Falcone!",
          "type": "div",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: rgba(200,200,200,0.5);
            }
            ",
        },
        "ref": null,
        "rendered": "
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: rgba(200,200,200,0.5);
            }
            ",
        "type": "style",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <header>
            <div
              className="title"
            >
              Finding Falcone!
            </div>
          </header>,
          <style>
            
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: rgba(200,200,200,0.5);
            }
            
          </style>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="title"
            >
              Finding Falcone!
            </div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Finding Falcone!",
              "className": "title",
            },
            "ref": null,
            "rendered": "Finding Falcone!",
            "type": "div",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: rgba(200,200,200,0.5);
            }
            ",
          },
          "ref": null,
          "rendered": "
            .title {
            font-size: 20px;
            }
            header {
                display: flex;
                justify-content: center;
                align-items: center;
                width: 100%;
                min-height: 70px;
                background: rgba(200,200,200,0.5);
            }
            ",
          "type": "style",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
